### PR TITLE
Adds host PID mode support for add-ons

### PR DESCRIPTION
**Pull request by frenck**
# Motivation

This PR adds support for add-ons running on PID mode: host. It allows for add-ons to run in the same PID namespace as the host OS.

This makes it possible for add-ons to e.g., perform monitoring/system profiling and other stuff. (e.g., Glances add-on).

### CODE REVIEW

File: hassio/addons/utils.py [hunk 1]
```diff
@@ -40,6 +40,10 @@ def rating_security(addon):
     if addon.host_network:
         rating += -1
 
+    # Insecure PID namespace
+    if addon.host_pid:
+        rating += -1
```
Comment (pvizeli) [inline hassio/addons/utils.py]: `-2` if the add-on run in same namespace, a attacker can access to host process.

**Comment by frenck** [comment]
I was thinking the same thing... protection mode enforcing this feature.
Since I was not sure, I've initially made this without it (and see the reaction).

I'll add protection.

File: hassio/addons/utils.py (hunk 1 above)
Comment (frenck) [reply hassio/addons/utils.py]: How about just setting the `rating = 1` just like with the docker API?

**Comment by frenck** [comment]
Tested locally and works like a charm.

image


```txt
root@local-glances:/$ ps faux
PID   USER     TIME  COMMAND
    1 root      0:19 {systemd} /sbin/init
    2 root      0:00 [kthreadd]
    4 root      0:00 [kworker/0:0H]
    6 root      0:00 [mm_percpu_wq]
    7 root      0:01 [ksoftirqd/0]
    8 root      0:01 [rcu_sched]
    9 root      0:00 [rcu_bh]
   10 root      0:00 [migration/0]
   11 root      0:00 [watchdog/0]
   12 root      0:00 [cpuhp/0]
   13 root      0:00 [cpuhp/1]
   14 root      0:00 [watchdog/1]
   15 root      0:00 [migration/1]
   16 root      0:00 [ksoftirqd/1]
   18 root      0:00 [kworker/1:0H]
   19 root      0:00 [kdevtmpfs]
   20 root      0:00 [netns]
   21 root      0:00 [rcu_tasks_kthre]
   22 root      0:00 [kauditd]
   24 root      0:00 [khungtaskd]
   25 root      0:00 [oom_reaper]
   26 root      0:00 [writeback]
   27 root      0:00 [kcompactd0]
   28 root      0:00 [ksmd]
   29 root      0:00 [khugepaged]
   30 root      0:00 [crypto]
   31 root      0:00 [kintegrityd]
   32 root      0:00 [kblockd]
   33 root      0:00 [ata_sff]
   34 root      0:00 [md]
   35 root      0:00 [edac-poller]
   36 root      0:00 [devfreq_wq]
   37 root      0:00 [watchdogd]
   40 root      0:01 [kswapd0]
   41 root      0:00 [ecryptfs-kthrea]
   84 root      0:00 [kthrotld]
   85 root      0:00 [acpi_thermal_pm]
   86 root      0:00 [scsi_eh_0]
   87 root      0:00 [scsi_tmf_0]
   88 root      0:00 [scsi_eh_1]
   89 root      0:00 [scsi_tmf_1]
   94 root      0:00 [ipv6_addrconf]
  103 root      0:00 [kstrp]
  120 root      0:00 [charger_manager]
  188 root      0:00 [mpt_poll_0]
  189 root      0:00 [mpt/0]
  192 root      0:00 [scsi_eh_2]
```

**Pull request merged**